installers/msi-language/S3Pixel/S3Pixel.cs

```diff
@@ -0,0 +1,51 @@
+using System;
+using GAPixel;
+using System.Net;
+using Microsoft.Deployment.WindowsInstaller;
+using System.IO;
+
+namespace S3Pixel
+{
+    public class CustomActions
+    {
+        [CustomAction]
+        public static ActionResult DownloadPixel(Session session)
+        {
+            session.Log("Begin download S3 pixel");
+            
+            string timeStamp = DateTime.Now.ToFileTime().ToString();
+            string tempDir = Path.Combine(Path.GetTempPath(), timeStamp);
+
+            session.Log("Creating temporary directory for S3 pixel file");
+            try
+            {
+                Directory.CreateDirectory(tempDir);
+            }
+            catch (Exception e)
+            {
+                string msg = string.Format("Could not create temp directory at: {0}, encountered exception: {1}", tempDir, e.ToString());
+                session.Log(msg);
+                RollbarReport.Error(msg);
+            }
+
+            string guid = GetInfo.GetUniqueId(session);
+            string pixelURL = string.Format("https://cli-msi.s3.amazonaws.com/pixel.txt?x-referrer={0}", guid);
+            string pixelFile = Path.Combine(tempDir, "pixel.txt");
```

Comment:
You shouldn't need to download to a physical file, see for example:

http://github.com/ActiveState/cli/blob/085e305011282f4f02d7d899f47b0449a2cbb7ce/installers/msi-language/GAPixel/GAPixelCA.cs#L95-L95